### PR TITLE
WT-12412 Model functions set_stable_timestamp and set_oldest_timestamp should match WT behavior

### PR DESCRIPTION
**Pull request by pmacko86**

### CODE REVIEW

File: test/model/test/model_basic/main.cpp [hunk 1]
```diff
@@ -1223,9 +1222,8 @@ test_model_oldest_wt(void)
     wt_model_assert(table, uri, key1, 40);
     wt_model_assert(table, uri, key1, 50);
 
-    /* FIXME-WT-12412: Return an error if the provided timestamp is older than the current one. */
-    /* Test moving the oldest timestamp backwards - this should fail silently. */
-    /* wt_model_set_oldest_timestamp_both(10); */
+    /* Test moving the oldest timestamp backwards - this should fail. */
+    wt_model_set_oldest_timestamp_both(10);
```
Comment (korteland) [inline test/model/test/model_basic/main.cpp:1226]: Is this call missing a `testutil_check(... == EINVAL)` wrapper?

File: test/model/test/model_checkpoint/main.cpp [hunk 2]
```diff
@@ -284,10 +284,8 @@ test_checkpoint_wt(void)
     wt_model_ckpt_assert(table, uri, "ckpt3", key2);
     wt_model_ckpt_assert(table, uri, "ckpt3", key3);
 
-    /*
-     * Test assigning the stable timestamp the same value again - this should not have any effect.
-     */
-    wt_model_set_stable_timestamp_both(60);
+    /* Test moving the stable timestamp backwards - this should fail. */
+    wt_model_set_stable_timestamp_both(50);
```
Comment (korteland) [inline test/model/test/model_checkpoint/main.cpp:288]: Same deal - `testutil_check`?

File: test/model/test/model_basic/main.cpp (hunk 1 above)
Comment (pmacko86) [reply test/model/test/model_basic/main.cpp:1226]: The definition of `wt_model_set_oldest_timestamp_both` includes an assertion that both WT and the model return the same return value.

File: test/model/test/model_checkpoint/main.cpp (hunk 2 above)
Comment (pmacko86) [reply test/model/test/model_checkpoint/main.cpp:288]: The same for `wt_model_set_stable_timestamp_both`.

File: test/model/test/model_basic/main.cpp (hunk 1 above)
Comment (korteland) [reply test/model/test/model_basic/main.cpp:1226]: Got it - thanks.

**Review by korteland — APPROVED** [review]
LGTM :-)

File: test/model/test/model_checkpoint/main.cpp [hunk 3]
```diff
@@ -157,22 +157,22 @@ test_checkpoint(void)
...
     txn1->commit(60, 60);
     txn2->commit(60, 65);
-    database.set_stable_timestamp(60);
+    testutil_check(database.set_stable_timestamp(60));
     model::kv_checkpoint_ptr ckpt3 = database.create_checkpoint("ckpt3");
     testutil_assert(table->get(ckpt3, key1) == value4);
     testutil_assert(table->get(ckpt3, key2) == value2); /* The old value. */
     testutil_assert(table->get(ckpt3, key3) == value3);
 
     /* Test moving the stable timestamp backwards - this should fail silently. */
```
Comment (etienneptl) [inline test/model/test/model_checkpoint/main.cpp:166]: It no longer fails "silently":
```suggestion
    /* Test moving the stable timestamp backwards - this should fail. */
```

**Review by etienneptl — APPROVED** [review]
LGTM, left one comment.
Do you have a scenario where the oldest is set before the stable is? This is the only scenario where we can have an oldest ts "greater" than the stable ts because it is unset.

**Comment by pmacko86** [comment]
> Do you have a scenario where the oldest is set before the stable is? This is the only scenario where we can have an oldest ts "greater" than the stable ts because it is unset.

Yes, in the test `model_basic`, in `test_model_oldest()` and `test_model_oldest_wt()`:

```
    /* Now try it with both the oldest and stable timestamps. */
    wt_model_set_oldest_timestamp_both(50);
    wt_model_set_stable_timestamp_both(55);
    database.restart();
    testutil_check(session->close(session, nullptr));
    testutil_check(conn->close(conn, nullptr));
    testutil_wiredtiger_open(opts, test_home.c_str(), ENV_CONFIG, nullptr, &conn, false, false);
    testutil_check(conn->open_session(conn, nullptr, nullptr, &session));
    testutil_assert(database.oldest_timestamp() == wt_get_oldest_timestamp(conn));
```

File: test/model/test/model_checkpoint/main.cpp (hunk 3 above)
Comment (pmacko86) [reply test/model/test/model_checkpoint/main.cpp]: Nice catch. Fixed in [2ca293c](https://github.com/wiredtiger/wiredtiger/pull/10545/commits/2ca293cc206a4fcd452d44580c27e201e2eeab10). Thanks!

**Comment by pmacko86** [comment]
Oh, I see, never mind. There are no such relative checks for the oldest and the stable timestamp, so I just added them in; that's a good suggestion. I just added them within this ticket as it's close enough to its scope; see [16aeda8](https://github.com/wiredtiger/wiredtiger/pull/10545/commits/16aeda82ee7d355bdf0b9f9a57bad3b8064576ff).

@etienneptl, @korteland, given that this is a non-trivial (albeit small) change, could you please review this last commit before I merge?

**Review by korteland — APPROVED** [review]
Still looks good.

**Review by etienneptl — APPROVED** [review]
LGTM! As always, thank you for taking the time to address all the comments/questions!

**Pull request merged**